rest/rest/src/main/java/io/syndesis/rest/v1/handler/connection/ConnectorHandler.java

```diff
@@ -141,4 +153,44 @@ public Kind resourceKind() {
         final Map<String, String> props) {
         return verifier.verify(connectorId, encryptionComponent.decrypt(props));
     }
+
+    @GET
+    @Path("/{id}/icon")
+    public Response getConnectorIcon(@NotNull @PathParam("id") final String connectorId) {
+        Connector connector = get(connectorId);
+        String connectorIcon = connector.getIcon();
+
+        // If there is no specified icon, return 404
+        if (connectorIcon == null) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+
+        // If the specified icon is a data URL, or a non-URL like value (e.g. font awesome class name), return 404
+        if (connectorIcon.startsWith("data:") || !connectorIcon.contains("/")) {
```

Comment:
Couldn't we serve the data URL by returning the (converted) data back to the client ?

Reply:
We could, but why?

Reply:
So that we can store the icon directly in the DB and do not have access the external service ? (e.g. allow to upload an icon in the UI instead of refering it to an exernal service)

Reply:
Data URLs are stored in the DB anyway in the `Connector.icon` field. We don't really need to convert them back to an image I don't think, the image source in the browser will just look like:

    <img src="data:image/gif;base64,R0lGODlhEAAOALMAAOazToeHh0tLS/7LZv/0jvb29t/f3//Ub//ge8WSLf/rhf/3kdbW1mxsbP//mf///yH5BAAAAAAALAAAAAAQAA4AAARe8L1Ekyky67QZ1hLnjM5UUde0ECwLJoExKcppV0aCcGCmTIHEIUEqjgaORCMxIC6e0CcguWw6aFjsVMkkIr7g77ZKPJjPZqIyd7sJAgVGoEGv2xsBxqNgYPj/gAwXEQA7" width="16" height="14" alt="embedded folder icon">